packages/astro/src/actions/index.ts

```diff
@@ -20,10 +20,17 @@ export default function astroActions({
 	fs?: typeof fsMod;
 	settings: AstroSettings;
 }): AstroIntegration {
+	let isActionsUsed = false;
+	let srcDir: URL;
 	return {
 		name: VIRTUAL_MODULE_ID,
 		hooks: {
 			async 'astro:config:setup'(params) {
+				isActionsUsed = actionsFileExists(fs, params.config.srcDir);
```

Comment:
It's not an integration. It's in core 🙃

Reply:
I have realized this check isn't full-proof though. To make it backwards compatible, we should check if the `server` export is present before booting up actions

Reply:
I know it's in core but we are actually inserting an integration in the astro config, that's what i meant. like instead of checking in each hook, check as early as possible

Reply:
Ah, perhaps we could. The issue is that `astro:server:setup` hook, which we _do_ want to mount to detect when actions are added in your project. If we conditionally add the whole integration, we can't check for this

Reply:
Ah I didn't know, current approach seems go then